agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -258,6 +272,80 @@ func (state *DockerTaskEngineState) RemoveENIAttachment(mac string) {
 	}
 }
 
+// GetAllEBSAttachments returns all the ebs volumes managed by ecs on the instance
+func (state *DockerTaskEngineState) GetAllEBSAttachments() []*apiresource.ResourceAttachment {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	return state.allEBSAttachmentsUnsafe()
+}
+
+func (state *DockerTaskEngineState) allEBSAttachmentsUnsafe() []*apiresource.ResourceAttachment {
+	var allEBSAttachments []*apiresource.ResourceAttachment
+	for _, v := range state.ebsAttachments {
+		allEBSAttachments = append(allEBSAttachments, v)
+	}
+	return allEBSAttachments
+}
+
+// GetAllPendingEBSAttachments returns all the ebs volumes managed by ecs on the instance that haven't sent a state change yet
+func (state *DockerTaskEngineState) GetAllPendingEBSAttachments() []*apiresource.ResourceAttachment {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	return state.allPendingEBSAttachmentsUnsafe()
+}
+
+func (state *DockerTaskEngineState) allPendingEBSAttachmentsUnsafe() []*apiresource.ResourceAttachment {
+	var allEBSAttachments []*apiresource.ResourceAttachment
+	for _, v := range state.ebsAttachments {
+		if !v.IsSent() {
+			allEBSAttachments = append(allEBSAttachments, v)
+		}
+	}
+	return allEBSAttachments
+}
+
+// AddEBSAttachment adds the ebs volume to state
+func (state *DockerTaskEngineState) AddEBSAttachment(ebsAttachment *apiresource.ResourceAttachment) {
+	if ebsAttachment == nil {
+		seelog.Debug("Cannot add empty ebs attachment information")
+		return
+	}
+	state.lock.Lock()
+	defer state.lock.Unlock()
+	volumeId := ebsAttachment.AttachmentProperties[apiresource.VolumeIdName]
+	if _, ok := state.ebsAttachments[volumeId]; !ok {
+		state.ebsAttachments[volumeId] = ebsAttachment
+	} else {
+		seelog.Debugf("Duplicate ebs attachment information: %v", ebsAttachment)
+	}
+}
+
+// RemoveEBSAttachment removes the ebs volume from state and stops managing
+func (state *DockerTaskEngineState) RemoveEBSAttachment(volumeId string) {
+	if volumeId == "" {
+		seelog.Debug("Cannot remove empty ebs attachment information")
+		return
+	}
+	state.lock.Lock()
+	defer state.lock.Unlock()
+	if _, ok := state.ebsAttachments[volumeId]; ok {
+		delete(state.ebsAttachments, volumeId)
```

Comment:
maybe we can add a log for successful deletion too (and attachment)

Reply:
ok!

Reply:
added a log statement for successful deletion